### PR TITLE
pppoe-server: T3006: Add range to regex generator

### PR DESCRIPTION
**Pull request by DmitriyEshenko**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3006

## Checklist:

- [X] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [X] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [X] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by DmitriyEshenko** [comment]
This PR depends on `range validator`, please cherrypick it before https://github.com/vyos/vyos-1x/commit/45d2429aa5d2ffafacdc5d9d00b7097169592427#diff-49e318d189379e8799aeb31525704e87adf363b6df13658e3e5704436c8e05b7

File: interface-definitions/service_pppoe-server.xml.in [hunk 1]
```diff
@@ -70,19 +70,25 @@
...
               <leafNode name="vlan-id">
                 <properties>
-                  <help>VLAN monitor for the automatic creation of vlans (user per vlan)</help>
-                  <constraint>
-                    <validator name="numeric" argument="--range 1-4096"/>
+                  <help>VLAN monitor for the automatic creation of single vlan</help>
+                  <valueHelp>
+                    <format>u32:1-4094</format>
+                    <description>VLAN monitor for the automatic creation of single vlan</description>
+                  </valueHelp>                  <constraint>
```
Comment (c-po) [inline interface-definitions/service_pppoe-server.xml.in]: oops - `<constraint>` moved into the wrong line.

File: interface-definitions/service_pppoe-server.xml.in [hunk 2]
```diff
@@ -70,19 +70,25 @@
...
               <leafNode name="vlan-range">
                 <properties>
-                  <help>VLAN monitor for the automatic creation of vlans (user per vlan)</help>
-                  <constraint>
-                    <regex>(409[0-6]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{0,2})-(409[0-6]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{0,2})</regex>
+                  <help>VLAN monitor for the automatic creation of vlans range</help>
+                  <valueHelp>
+                    <format>start-end</format>
+                    <description>VLAN monitor range for the automatic creation of vlans (e.g. 1-4094)</description>
+                  </valueHelp>                  <constraint>
```
Comment (c-po) [inline interface-definitions/service_pppoe-server.xml.in]: oops - `<constraint>` moved into the wrong line.

File: src/conf_mode/service_pppoe-server.py [hunk 3]
```diff
@@ -73,6 +80,13 @@ def generate(pppoe):
     if not pppoe:
         return None
 
+    # Generate special regex for dynamic interfaces
+    for iface in pppoe['interface']:
```
Comment (c-po) [inline src/conf_mode/service_pppoe-server.py]: Please use the more readable format:

```python
for iface, iface_options in pppoe['interface'].items():
    if 'vlan_range' in iface_options:
        ...
```

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/service_pppoe-server.xml.in (hunk 1 above)
Comment (DmitriyEshenko) [reply interface-definitions/service_pppoe-server.xml.in]: Sorry for this mistake. Done.

File: interface-definitions/service_pppoe-server.xml.in (hunk 2 above)
Comment (DmitriyEshenko) [reply interface-definitions/service_pppoe-server.xml.in]: Done.

File: src/conf_mode/service_pppoe-server.py (hunk 3 above)
Comment (DmitriyEshenko) [reply src/conf_mode/service_pppoe-server.py]: Done

**Pull request merged**